lib/context_adapter_helper.js

```diff
@@ -0,0 +1,724 @@
+   * @param {string} servicePath Service Path header to be used in the request
+   * @param {object} payload Payload to be included in the request
+   * @param {function} callback Callback to be called informing of any errors or returning
+   *  the obtained response
+   */
+  function sendUpdateContext(service, servicePath, payload, callback) {
+    request({
+      method: 'POST',
+      uri: 'http://' + caConfig.CB_HOST + ':' + caConfig.CB_PORT + caConfig.CB_PATH + '/updateContext',
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'Fiware-Service': service || caConfig.SERVICE,
+        'Fiware-ServicePath': servicePath || caConfig.SERVICE_PATH
+      },
+      json: true,
+      body: payload
+    }, callback);
+  }
+
+  /**
+   * Indicates if certain attribute is included in an attributes array
+   * @param attributes The attributes array
+   * @param attributeName The attribute name to find
+   * @return {boolean} The result of the search
+   */
+  function isAttributeIncluded(attributes, attributeName) {
+    for (var i = 0; i < attributes.length; i++) {
+      if (attributes[i].name && attributes[i].name === attributeName) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Checks that an updateContext request sent to the Context Adapter is well-formed
+   * @param {object} request The request sent to the Context Adapter
+   * @param {function} callback Callback to be called with the result of the check
+   */
+  function checkUpdateContextRequest(request, callback) {
+    /*
+      A valid updateContext payload is one such as the following one:
+        {
+          contextElements: [
+            {
+              id: <device-id>,
+              type: caConfig.BUTTON_ENTITY.TYPE,
+              isPattern: false,
+              attributes: [
+                {
+                  name: caConfig.BUTTON_ENTITY.CA_EXTERNAL_ID_ATTR_NAME,
+                  type: 'string',
+                  value: '<aux-external-id>'
+                },
+                {
+                  name: caConfig.BUTTON_ENTITY.CA_SERVICE_ID_ATTR_NAME,
+                  type: 'string',
+                  value: '<aux-service-id>'
+                },
+                {
+                  name: caConfig.BUTTON_ENTITY.CA_OPERATION_ACTION_ATTR_NAME,
+                  type: 'string',
+                  value: 'S | C'
+                },
+                {
+                  name: caConfig.BUTTON_ENTITY.CA_OPERATION_EXTRA_ATTR_NAME,
+                  type: 'string',
+                  value: '<aux-op-extra>'
+                },
+                {
+                  name: caConfig.BUTTON_ENTITY.CA_OPERATION_STATUS_ATTR_NAME,
+                  type: 'string',
+                  value: '<aux-op-status>'
+                }
+              ]
+            }
+          ],
+          updateAction: 'UPDATE'
+        }
+     */
+    if (request.payload && request.payload.contextElements &&
```

Comment:
If you are really going to check that many things, consider using JSON Schema with the corresponding library (e.g.: [revalidator](https://github.com/flatiron/revalidator) ).